src/core/parser/parser.py

```diff
@@ -21,98 +17,87 @@ def __init__(self, **kwargs):
+            return pythonTodoParser()
+        elif language == "HTML":
+            return htmlTodoParser()
+        else:
+            raise ValueError(language)
             
 
-# This should be a different class, will refract code later
-def _getTODOPython(filename):
-    """
-    This function extract the to dos for a given file
-    This will return the todo in this format 
-    {"filepath":"path/to/file.py","todo":[(lineno,message)]}
-    """
-    # There are two ways todo can be in a file after # or inside """
-    # Above line will also be returned by parser #logic :P 
-    todoList = list()
-    todoDict = dict()
-    with open(filename) as f:
-        lines = f.readlines()
-        for i in range(0,len(lines)):
-            if "#" in lines[i]: # since comment can be after the code as well
-                stripedLine = lines[i][lines[i].find("#"):].strip().lower()
-                # TODO will modify the logic to adapt variants of todo
-                if "todo" in stripedLine:
-                    todoList.append( (i+1,stripedLine[stripedLine.find("todo")+4:].strip()) )
-    if len(todoList) > 0:
-        todoDict["filepath"] = filename
-        todoDict["todo"] = todoList
-        return todoDict
-    else:
-        return None
-
-# This should be a different class, will refract code later, ignore code repeatition for now
-# As we will be adding support of other languages as well
-def _getTODOHTML(filename):
-    """
-    This function extract the to dos for a given file
-    This will return the todo in this format 
-    {"filepath":"path/to/file.html","todo":[(lineno,message)]}
-    """
-    # There is one wat I know todo can be added to a html file i.e inside <!--
-    # Above line will also be returned by parser #logic :P 
-    todoList = list()
-    todoDict = dict()
-    with open(filename) as f:
-        lines = f.readlines()
-        for i in range(0,len(lines)):
-            if "<!--" in lines[i]: # since comment can be after the code as well
-                stripedLine = lines[i][lines[i].find("<!--"):].strip().lower() # Value in find() can be replaced by the character of language used for commenting
-                # TODO will modify the logic to adapt variants of todo
-                if "todo" in stripedLine:
-                    todoList.append( (i+1,stripedLine[stripedLine.find("todo")+4:].strip()) )
-    if len(todoList) > 0:
-        todoDict["filepath"] = filename
-        todoDict["todo"] = todoList
-        return todoDict
-    else:
-        return None
-
-# This will return the appropriate parser
-def get_parser(language):
-    if language == "Python": 
-        return _getTODOPython
-    elif language == "HTML":
-        return _getTODOHTML
-    else:
-        raise ValueError(language)
-
-# Driver Code to test parser
-# def parser_test():
-#     filename = "./parser.py"
-#     parser_obj = Parser()
-#     print(parser_obj.getTODOPython(filename))
-
-# parser_test()
+class pythonTodoParser:
```

Comment:
Split class into separate files and capitalise 'P'

Reply:
Okay